jsx/CSSGrid.js

```diff
@@ -0,0 +1,45 @@
+import Card from 'Card';
+
+/**
+ * Create a three column grid of cards using a CSS grid.
+ * The grid is densely populated (if there's a hole in the
+ * grid and a later card fits in it, it will be added to it.)
+ *
+ * @param {array} props - Cards to add to the grid.
+ *
+ * @return {object} - A React component for a CSS grid of cards
+ */
+function CSSGrid(props) {
+  const grid = {
+    display: 'grid',
+    gridTemplateColumns: '33% 33% 33%',
+    gridAutoFlow: 'row dense',
+    gridRowGap: '1em',
+    rowGap: '1em',
+  };
+
+  let cards = props.Cards.map((value, idx) => {
+    let cardID = 'card' + idx;
+
+    let style= {};
+    if (value.Width) {
+      style.gridColumnEnd = 'span ' + value.Width;
+    }
+    if (value.Height) {
+      style.gridRowEnd = 'span ' + value.Height;
+    }
+    if (value.Order) {
+      style.order = value.Order;
+    }
+
+    style.alignSelf = 'stretch';
```

Comment:
Hey @driusan, one sugge
```
const style = {
     ...(value.width ? {gridColumnEnd: `span $value.Width`} : {}),
     ...(value.Height ? {gridRowEnd: `span $value.Height`} : {}),
     ...(value.Order ? {order: value.Order} : {}),
     alignSelf: 'strech'
   };
```
You could have it like that instead of the if statements for adding to the style object. Thought it might be worth mentioning. You may not like.

Reply:
This file is just pulled in from #6093, it should go there (and I find it order to much read with the ternary operator, I don't think it being const makes affects anything enough that we should sacrifice readability.)

Reply:
@driusan best to ignore my suggestion if it's harder to read.